tool/matomo/classes/client.php

```diff
@@ -0,0 +1,210 @@
+     */
+    public function get_siteid_from_url(string $url = ''): int {
+        global $CFG;
+
+        $request = $this->build_request();
+        $request['method'] = 'SitesManager.getSitesIdFromSiteUrl';
+        $request['url'] = !empty($url) ? $url : $CFG->wwwroot;
+        $rawresponsebody = $this->post($this->url, $request);
+        $response = $this->validate_response_body($rawresponsebody, $request);
+        $response = is_array($response) ? reset($response) : new stdClass();
+
+        return !empty($response->idsite) ? $response->idsite : 0;
+    }
+
+    /**
+     * Get all registered urls for this siteid.
+     *
+     * @param int $siteid
+     * @return array
+     */
+    public function get_urls_from_siteid(int $siteid): array {
+        $request = $this->build_request();
+        $request['method'] = 'SitesManager.getSiteUrlsFromId';
+        $request['idSite'] = $siteid;
+        $rawresponsebody = $this->post($this->url, $request);
+        $response = $this->validate_response_body($rawresponsebody, $request);
+        return is_array($response) ? $response : [];
+    }
+
+    /**
+     * Create a site at Matomo instance.
+     *
+     * @param string $sitename
+     * @param string[] $urls
+     * @param string $timezone
+     * @param string $currency
+     * @return int Site id on success, or 0 on failure.
+     * @throws Exception
+     */
+    public function add_site(string $sitename = '', array $urls = [], string $timezone = '', string $currency = ''): int {
+        global $SITE;
+
+        $request = $this->build_request();
+        $request['method'] = 'SitesManager.addSite';
+        $request['siteName'] = !empty($sitename) ? $sitename : $SITE->fullname;
+        $request['timezone'] = !empty($timezone) ? $timezone : 'Europe/London';
+        $request['currency'] = !empty($currency) ? $currency : 'GBP';
+        $request = array_merge($request, $this->build_urls_for_request($urls));
+        $rawresponse = $this->post($this->url, $request);
+        $responsebody = $this->validate_response_body($rawresponse, $request);
+
+        return !empty($responsebody->value) ? $responsebody->value : 0;
+    }
+
+    /**
+     * Update site at Matomo instance.
+     *
+     * @param int $siteid
+     * @param string $sitename
+     * @param array $urls
+     * @param string $timezone
+     * @param string $currency
+     * @return int
+     * @throws Exception
+     */
+    public function update_site(
+            int $siteid,
+            string $sitename = '',
+            array $urls = [],
+            string $timezone = '',
+            string $currency = ''
+    ): int {
+        global $SITE;
+
+        $request = $this->build_request();
+        $request['method'] = 'SitesManager.updateSite';
+        $request['idSite'] = $siteid;
+        $request['siteName'] = !empty($sitename) ? $sitename : $SITE->fullname;
+        $request['timezone'] = !empty($timezone) ? $timezone : 'Europe/London';
```

Comment:
shouldn't this default to the moodles default timezone? eg core_date::set_default_server_timezone() or similar